volatility/framework/plugins/windows/getsids.py

```diff
@@ -0,0 +1,257 @@
+                                 base_config_path = self.config_path,
+                                 layer_name = self.config['primary'],
+                                 symbol_table = self.config['nt_symbols'],
+                                 hive_offsets = None):
+        
+            try:
+                for subkey in hive.get_key(key).get_subkeys():
+                    sid = str(subkey.get_name())
+                    path = ""
+                    for node in subkey.get_values():
+                        try:
+                            value_node_name = node.get_name() or "(Default)"
+                        except (exceptions.InvalidAddressException, RegistryFormatException) as excp:
+                            continue
+                        try:
+                            value_data = node.decode_data()
+                            if isinstance(value_data, int):
+                                value_data = format_hints.MultiTypeData(value_data, encoding = 'utf-8')
+                            elif RegValueTypes.get(node.Type) == RegValueTypes.REG_BINARY:
+                                value_data = format_hints.MultiTypeData(value_data, show_hex = True)
+                            elif RegValueTypes.get(node.Type) == RegValueTypes.REG_MULTI_SZ:
+                                value_data = format_hints.MultiTypeData(value_data,
+                                                                        encoding = 'utf-16-le',
+                                                                        split_nulls = True)
+                            else:
+                                value_data = format_hints.MultiTypeData(value_data, encoding = 'utf-16-le')
+                            if value_node_name == val:
+                                path = str(value_data).replace('\\x00', '')[:-1]
+                                user = ntpath.basename(path)
+                                sids[sid] = user
+                        except (ValueError, exceptions.InvalidAddressException, RegistryFormatException) as excp:
+                            continue
+            except KeyError:
+                continue
+
+        return sids
+
+    def _generator(self, procs):
+
+        user_sids = self.lookup_user_sids()
+
+        for task in procs:
+            token = task.Token.dereference().cast("_TOKEN")
+
+            if not token:
+                yield (0, [int(task.UniqueProcessId),
+                                 str(task.ImageFileName),
+                                 "Token unreadable",
+                                 ""])
+                continue
+
+            for sid_string in token.get_sids():
+                if sid_string in well_known_sids:
+                    sid_name = well_known_sids[sid_string]
+                elif sid_string in getservicesids.servicesids:
+                    sid_name = getservicesids.servicesids[sid_string]

```

Comment:
This would be a perfect place to load the JSON rather than loading another plugin file.